lib/foreman_maintain/cli/backup_command.rb

```diff
@@ -0,0 +1,148 @@
+          option '--skip-pulp-content', :flag, 'Do not backup Pulp content'
+          option '--preserve-directory', :flag, 'Do not create a time-stamped subdirectory'
+          option '--split-pulp-tar', 'SPLIT_SIZE',
+                 'Split pulp data into files of a specified size, i.e. (100M, 50G). ' \
+                 "See '--tape-length' in 'info tar' for all sizes" do |size|
+            self.class.valid_tape_size(size)
+          end
+          option '--incremental', 'PREVIOUS_BACKUP_DIR',
+                 'Backup changes since previous backup' do |dir|
+            unless File.directory?(dir)
+              raise ArgumentError, "Previous backup directory does not exist: #{dir}"
+            end
+            dir
+          end
+          proxy_name = ForemanMaintain.detector.feature(:downstream) ? 'Capsule' : 'Foreman Proxy'
+          option '--features', 'FEATURES',
+                 "#{proxy_name} features to include in the backup. " \
+                     'Valid features are tftp, dns, dhcp, openscap, and all.', :multivalued => true
+        end
+        # rubocop:enable  Metrics/MethodLength
+
+        def valid_tape_size(size)
+          begin
+            ForemanMaintain.detector.feature(:tar).validate_volume_size(size)
+          rescue ForemanMaintain::Error::Validation => e
+            raise ArgumentError, e.message
+          end
+          size
+        end
+      end
+
+      private
+
+      def rescue_scenario(options, strategy)
+        Scenarios::BackupRescueCleanup.new({
+          :backup_dir => backup_dir,
+          :strategy => strategy,
+          :preserve_dir => preserve_directory?
+        }.merge(options))
+      end
+
+      def backup_scenario(options, strategy)
+        Scenarios::Backup.new({
+          :backup_dir => backup_dir,
+          :strategy => strategy,
+          :preserve_dir => preserve_directory?,
+          :proxy_features => features,
+          :tar_volume_size => split_pulp_tar,
+          :skip_pulp_content => skip_pulp_content?,
+          :incremental_dir => incremental
+        }.merge(options))
+      end
+
+      def final_message
+        if runner.quit?
+          if preserve_directory?
+            puts "Backup didn't finish. Incomplete backup is preserved in: #{backup_dir}"
+          else
+            puts "Backup didn't finish. Incomplete backup was removed."
+          end
+        else
+          puts "**** BACKUP Complete, contents can be found in: #{backup_dir} ****"
+        end
+      end
+    end
+
+    class OnlineBackupCommand < Base
+      include BackupCommon
+      interactive_option
+      common_backup_options
+
+      def execute
+        perform_backup(:online)
+      end
+    end
+
+    class OfflineBackupCommand < Base
+      include BackupCommon
+      interactive_option
+      common_backup_options
+      option '--logical-db-backup', :flag, 'Also dump full database schema during offline backup'
```

Comment:
Do you still plan on changing this to `--include-online`?

Reply:
I've renamed the option, experimenting with subdir. Will add it if it is a simple fix

Reply:
I found out we do only online dump of the DBs during 'logical' backup having the configs and metadata shared with offline. We would have to do the common data twice or adding symlinks to have two separate restorable backups. Not sure if it is worth it. We can do it as a separate PR if there is a demand for it.

Reply:
Sure, that's what I did in https://github.com/theforeman/foreman_maintain/pull/164/commits/3360f8c7c2bd67e17a2ebb1271ba3142276d3677